tests/filecheck/projects/riscv-backend-paper/bottom_up.mlir

```diff
@@ -593,7 +593,7 @@ func.func public @pooling_nchw_max_d1_s2_3x3(
           #snitch_stream.stride_pattern<ub = [128], strides = [8]>
         ]
       } ins(%X_1 : !riscv.reg) outs(%Y_1 : !riscv.reg) {
-      ^0(%x : !stream.readable<!riscv.freg<ft0>>, %0 : !stream.writable<!riscv.freg<ft1>>):
+      ^0(%x : !snitch.readable<!riscv.freg<ft0>>, %0 : !snitch.writable<!riscv.freg<ft1>>):
```

Comment:
What was the decision process for when to use `snitch` and when to use `memref_stream`. This file for example uses both

Reply:
This is probably just me not knowing much about this side of xDSL, but in this PR some cases of `!stream.(readable|writable)` are changed to `!memref_stream....` and some are changed to `!snitch....` and I couldn't really see a pattern. As a result I feel I can't really review this at the moment (though I wasn't pinged for a review (maybe for this reason)).

Reply:
it's entirely determined by the operation enclosing the block in question. In this case it's a snitch_stream.streaming region, in others it's memref_stream.streaming region.